jetty-openid/src/main/java/org/eclipse/jetty/security/openid/OpenIdCredentials.java

```diff
@@ -158,9 +158,9 @@ public boolean isExpired()
         if (sections.length != 3)
             throw new IllegalArgumentException("JWT does not contain 3 sections");
 
-        Base64.Decoder decoder = Base64.getDecoder();
-        String jwtHeaderString = new String(decoder.decode(sections[0]), StandardCharsets.UTF_8);
-        String jwtClaimString = new String(decoder.decode(sections[1]), StandardCharsets.UTF_8);
+        Base64.Decoder decoder = Base64.getUrlDecoder();
+        String jwtHeaderString = new String(decoder.decode(padJWTSection(sections[0])), StandardCharsets.UTF_8);
+        String jwtClaimString = new String(decoder.decode(padJWTSection(sections[1])), StandardCharsets.UTF_8);
```

Comment:
This seems quite odd.
This needs test cases to demonstrate this issue.
Can you generalize this decoder behavior into a new class and create unit tests for it?

Example:

``` java
public class CredentialsDecoder
{
  private static final Base64.Decoder decoder = Base64.getUrlDecoder();
  public static String decode(String section) 
  {
     return new String(decoder.decode(padJWTSection(section)), StandardCharsets.UTF_8);
  }
}
```

``` java
import java.util.stream.Stream;

import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;

public class CredentialsDecoderTest
{
    public static Stream<Arguments> examples()
    {
        return Stream.of(
            Arguments.of("THISINPUT_CHANGEME", "SHOULDPRODUCETHIS_CHANGEME"),
            Arguments.of("THISOTHERINPUT_CHANGEME", "SHOULDPRODUCETHISOTHER_CHANGEME")
        );
    }

    @ParameterizedTest
    @MethodSource("examples")
    public void testDecode(String input, String expected)
    {
        String actual = CredentialsDecoder.decode(input);
        assertThat(actual, is(expected));
    }
}
```

Reply:
This is being addressed in PR #4166.

Reply:
This is being resolved in PR #4166.